raiden_contracts/data/source/services/ServiceRegistry.sol

```diff
@@ -249,8 +253,13 @@ contract ServiceRegistry is Utils, ServiceRegistryConfigurableParameters {
         uint256 amount = current_price();
         require(_limit_amount >= amount, "not enough limit");
 
+        bool new_joiner = false;
+
         // Extend the service position.
         uint256 valid_till = service_valid_till[msg.sender];
+        if (valid_till == 0) { // a first time joiner
+            new_joiner = true;
```

Comment:
Why not just do a `ever_made_deposits.push(msg.sender);` here and skip the `new_joiner` variable completely?

Reply:
That sounds better.  I'll do this.